kratos/utilities/volume_inside_voxel_utility.cpp

```diff
@@ -0,0 +1,382 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ariadna Cortés
+//
+
+//Project includes
+#include "volume_inside_voxel_utility.h"
+
+namespace Kratos {
+
+    double VolumeInsideVoxelUtility::NodesApproximation(const GeometryType& rVoxel) {
+        double volume = 0;
+        PointsArrayType nodes = rVoxel.Points();
+        for (int i = 0; i < nodes.size(); i++) {
+            if (nodes[i].GetSolutionStepValue(DISTANCE) > 0) {
+                volume+=(1.0/nodes.size()); 
+            } 
+        }
+        return volume;
+    }
+
+    /***********************************************************************************
+     **********************************************************************************/
+
+    double VolumeInsideVoxelUtility::EdgesApproximation(const GeometryType& rVoxel) {
+        double volume = 0;
+        GeometryArrayType Edges = rVoxel.GenerateEdges();
+        PointsArrayType nodes = rVoxel.Points();
+        for (int i = 0; i < Edges.size(); i++) {
+            PointsArrayType ends = Edges[i].Points();
+            if(ends[0].GetSolutionStepValue(DISTANCE) > 0 && ends[1].GetSolutionStepValue(DISTANCE) > 0) {
+                volume+=1.0/Edges.size();
+            } else if(
+                ends[0].GetSolutionStepValue(DISTANCE) > 0 && ends[1].GetSolutionStepValue(DISTANCE) < 0 || 
+                ends[0].GetSolutionStepValue(DISTANCE) < 0 && ends[1].GetSolutionStepValue(DISTANCE) > 0 ) {
+                volume+=1.0/(Edges.size()*2);
+            }
+        }
+        return volume;
+    }
+
+    /***********************************************************************************
+     **********************************************************************************/
+
+    double VolumeInsideVoxelUtility::EdgesPortionApproximation(
+        const GeometryType& rVoxel,  
+        const GeometryArrayType& rTriangles     
+    ) {
+        double volume = 0;
+        GeometryArrayType Edges = rVoxel.GenerateEdges();
+        std::vector<double> Distances;
```

Comment:
```suggestion
        std::vector<double> distances;
```
Please stick to the style guide.

Reply:
Besides, you know the size a priori so you can allocate the space required and avoid the `push_back` below.